### PR TITLE
Turn repo into a multi-module maven project

### PR DESCRIPTION
**Pull request by rmetzger**

### CODE REVIEW

File: flink-ecosystem-frontend/pom.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                            <nodeVersion>v10.9.0</nodeVersion>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>npm install</id>
+                        <goals>
+                            <goal>npm</goal>
+                        </goals>
+                        <configuration>
+                            <arguments>install --cache-max=0 --no-save</arguments>
```
Comment (sorahn) [inline flink-ecosystem-frontend/pom.xml]: `--production` too, so anyone running the build doesn't get the dev tools.

Comment (rmetzger) [reply flink-ecosystem-frontend/pom.xml]: Ok. Will add.

**Comment by sorahn** [comment]
If you change the `.travis.yml` file to point to `flink-ecosystem-frontend` it should pass the build again.

Then we can figure out later how to make the whole thing work in docker.

**Comment by klion26** [comment]
checked the backend files, LGTM

**Comment by rmetzger** [comment]
How about merging this as-is, and then incrementally do changes?

**Pull request merged**